Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -1147,15 +1145,27 @@ void MediaSource::notifyElementUpdateMediaState() const
     mediaElement()->updateMediaState();
 }
 
-void MediaSource::updateBufferedIfNeeded()
+void MediaSource::sourceBufferBufferedChanged()
+{
+    updateBufferedIfNeeded();
+}
+
+void MediaSource::updateBufferedIfNeeded(bool force)
 {
-    if (m_buffered && m_activeSourceBuffers->length() && std::all_of(m_activeSourceBuffers->begin(), m_activeSourceBuffers->end(), [](auto& buffer) { return !buffer->isBufferedDirty(); }))
+    if (!force && m_activeSourceBuffers->length() && std::all_of(m_activeSourceBuffers->begin(), m_activeSourceBuffers->end(), [](auto& buffer) { return !buffer->isBufferedDirty(); }))
         return;
 
-    m_buffered = makeUnique<PlatformTimeRanges>();
     for (auto& sourceBuffer : *m_activeSourceBuffers)
         sourceBuffer->setBufferedDirty(false);
 
+    auto buffered = WTFMove(m_buffered);
```

Comment:
Can buffered be a PlatformTimeRanges instead of a unique_ptr?

Reply:
> Can buffered be a PlatformTimeRanges instead of a unique_ptr?

That would be changing the API from SourceBufferPrivateClient ; I guess we could. But there are implementation that return nullptr right now.

Reply:
such as `std::unique_ptr<PlatformTimeRanges> RemoteMediaSourceProxy::buffered()` however, I am making changes in that code soon, following which it can never be null

Reply:
On second thoughts, MediaSource::buffered() changes a lot and often, you probably want to avoid anything taking a reference and not make a copy first. Particularly as there's re-entrancy everywhere in that code. You never know if an object won't get modified from under you.